activation/activation.go

```diff
@@ -312,33 +306,30 @@ func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
 	// we've completed the sequential work, now before publishing the atx,
 	// we need to provide number of atx seen in the epoch of the positioning atx.
 	posEpoch := b.posLayerID.GetEpoch(b.layersPerEpoch)
-	activeIds, err := b.activeSet.ActiveSetSize(posEpoch)
-	if err != nil && !posEpoch.IsGenesis() {
-		return err
-	}
-	view, err := b.mesh.GetOrphanBlocksBefore(b.mesh.LatestLayer())
-	if err != nil {
-		return err
+	pubEpoch := b.challenge.PubLayerIdx.GetEpoch(b.layersPerEpoch) // can be 0 in first epoch
+	targetEpoch := pubEpoch + 1
+
+	viewLayer := pubEpoch.FirstLayer(b.layersPerEpoch)
+	var view []types.BlockID
+	if viewLayer > 0 {
+		var err error
+		view, err = b.mesh.GetOrphanBlocksBefore(viewLayer)
+		if err != nil {
+			return fmt.Errorf("failed to get current view: %v", err)
+		}
 	}
-	atx := types.NewActivationTxWithChallenge(*b.challenge, b.coinbaseAccount, activeIds, view, b.nipst)
-	activeSetSize, err := b.db.CalcActiveSetFromView(atx) // TODO: remove this assertion to improve performance
 
-	if err != nil && !atx.TargetEpoch(b.layersPerEpoch).IsGenesis() {
-		b.log.Warning("empty active set size found! len(view): %d, view: %v", len(atx.View), atx.View)
+	activeSetSize, err := b.db.CalcActiveSetFromView(view, pubEpoch)
```

Comment:
shouldn't we assert that active set size is > 0

Reply:
fixed